packages/j/joltphysics/xmake.lua

```diff
@@ -74,9 +76,75 @@ package("joltphysics")
         if package:config("shared") then
             package:add("defines", "JPH_SHARED_LIBRARY")
         end
+        if package:is_arch("i386", "x86", "x64", "x86_64") then
+            -- add instruction sets (from https://github.com/jrouwe/JoltPhysics/blob/4cd52055e09160affcafa557b39520331bf0d034/Jolt/Jolt.cmake#L602)
+            if package:has_tool("cxx", "cl") then
+                if package:config("inst_avx512") then
+                    package:add("cxxflags", "/arch:AVX512")
+                elseif package:config("inst_avx2") then
+                    package:add("cxxflags", "/arch:AVX2")
+                elseif package:config("inst_avx") then
+                    package:add("cxxflags", "/arch:AVX2")
+                end
+            else
+                if package:config("inst_avx512") then
+                    package:add("cxxflags", "-mavx512f", "-mavx512vl", "-mavx512dq", "-mavx2", "-mbmi", "-mpopcnt", "-mlzcnt", "-mf16c")
```

Comment:
These instructions are used in the header files of this library? 

We don't need to export these flags to the user if these flags are only used in the obj file of the library itself.

Reply:
Yes they are public in https://github.com/jrouwe/JoltPhysics/blob/4cd52055e09160affcafa557b39520331bf0d034/Jolt/Jolt.cmake
and it breaks tests to not have them on MinGW macOS

Reply:
Another possibility is to only export the mandatory flag (-mfpmath=sse) and make these flags optionnal (with a package rule to add them)

Reply:
> Yes they are public in https://github.com/jrouwe/JoltPhysics/blob/4cd52055e09160affcafa557b39520331bf0d034/Jolt/Jolt.cmake and it breaks tests to not have them on MinGW macOS

it works on my mingw macOS without these flags.

```console
patching /Users/ruki/.xmake/packages/j/joltphysics/v5.0.0/3ce7b350feaf4b76ac763253a0d392d9/lib/pk
gconfig/joltphysics.pc ..
> /usr/local/opt/mingw-w64/bin/x86_64-w64-mingw32-g++ -c -m64 -std=c++17 -DJPH_OBJECT_LAYER_BITS=
16 -DJPH_PROFILE_ENABLED -DJPH_DEBUG_RENDERER -isystem /Users/ruki/.xmake/packages/j/joltphysics/
v5.0.0/3ce7b350feaf4b76ac763253a0d392d9/include -o /var/folders/32/w9cz0y_14hs19lkbs6v6_fm80000gn
/T/.xmake501/240615/_2DD3DD81DB004500800FD9CE02464F00.o /var/folders/32/w9cz0y_14hs19lkbs6v6_fm80
000gn/T/.xmake501/240615/_11B43A0BC3BB4F32B7835AFE42E6E76D.cpp
checking for /usr/local/opt/mingw-w64/bin/x86_64-w64-mingw32-g++ ... ok
checking for flags (-fdiagnostics-color=always) ... ok
> x86_64-w64-mingw32-g++ "-fdiagnostics-color=always" "-m64"
> /usr/local/opt/mingw-w64/bin/x86_64-w64-mingw32-g++ -o /var/folders/32/w9cz0y_14hs19lkbs6v6_fm8
0000gn/T/.xmake501/240615/_2DD3DD81DB004500800FD9CE02464F00.b /var/folders/32/w9cz0y_14hs19lkbs6v
6_fm80000gn/T/.xmake501/240615/_2DD3DD81DB004500800FD9CE02464F00.o -m64 -L/Users/ruki/.xmake/pack
ages/j/joltphysics/v5.0.0/3ce7b350feaf4b76ac763253a0d392d9/lib -lJolt
> checking for c++ includes(Jolt/Jolt.h, Jolt/Physics/PhysicsSystem.h)
> checking for c++ links(Jolt)
> checking for c++ snippet(test)
  => install joltphysics v5.0.0 .. ok
```

> they are public in https://github.com/jrouwe/JoltPhysics/blob/4cd52055e09160affcafa557b39520331bf0d034/Jolt/Jolt.cmake

I don't see them being exported to .cmake import files, just configured for compiling libraries.

https://github.com/jrouwe/JoltPhysics/blob/4cd52055e09160affcafa557b39520331bf0d034/Jolt/Jolt.cmake#L619

Reply:
That's because you tested in 64bits which enables SSE by default, it fails in 32bits mode as you can see on the CI.

As for the cmake import files, they were just added very recently: https://github.com/jrouwe/JoltPhysics/commit/702a16b86b15d376837350e2e743336c216e8622

Reply:
ok